### PR TITLE
Issue #13668: finish coding package for modules wihtout tokens

### PR DESCRIPTION
**Pull request by romani**
part for Issue #13668

changes are: 
- due re-ordering of properties in javadoc and xdoc 
- usage of capital first letter in some description.

### CODE REVIEW

**Comment by romani** [comment]
It is general GitHub CI instabilities, restarted

**Comment by rnveach** [comment]
That includes link check?
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/20733/workflows/e41d4538-255a-44f9-b626-2e5825bcb5f6/jobs/381240
> `> <td><i><a href="property_types.html#subset_of_tokens_TokenTypes">../../property_types.html#subset_of_tokens_TokenTypes</a>: doesn't exist.</i></td></tr>`

File: src/xdocs/checks/coding/multiplestringliterals.xml [hunk 1]
```diff
@@ -32,39 +32,25 @@
...
-              </td>
+              <td>Specify the maximum number of occurrences to allow without generating a warning.</td>
               <td><a href="../../property_types.html#int">int</a></td>
               <td><code>1</code></td>
               <td>3.5</td>
             </tr>
+            <tr>
+              <td>ignoreOccurrenceContext</td>
+              <td>Specify token type names where duplicate strings are ignored even if they don't match ignoredStringsRegexp. This allows you to exclude syntactical contexts like annotations or static initializers from the check.</td>
+              <td><a href="../../property_types.html#subset of tokens TokenTypes">subset of tokens TokenTypes</a></td>
```
Comment (rnveach) [inline src/xdocs/checks/coding/multiplestringliterals.xml:43]: Not a valid link.
https://checkstyle.org/property_types.html#subset%20of%20tokens%20TokenTypes

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/xdocs/checks/coding/multiplestringliterals.xml (hunk 1 above)
Comment (rnveach) [reply src/xdocs/checks/coding/multiplestringliterals.xml:43]: Should be similar to https://checkstyle.org/checks/coding/finallocalvariable.html#FinalLocalVariable

Comment (rnveach) [reply src/xdocs/checks/coding/multiplestringliterals.xml:43]: Bad view: https://checkstyle.org/checks/coding/multiplestringliterals.html#MultipleStringLiterals

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs/checks/coding/multiplestringliterals.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/checks/coding/multiplestringliterals.xml]: @rnveach , I removed update on this module from this PR, we will deal with MultipleStringLiterals separately

Comment (romani) [reply src/xdocs/checks/coding/multiplestringliterals.xml]: moved to https://github.com/checkstyle/checkstyle/pull/13721

**Comment by romani** [comment]
Cirrus was restarted

**Comment by rnveach** [comment]
Cirrus didn't restart.

**Comment by rnveach** [comment]
@romani Now that we have identified the macro as having a bug, is there anyway you can prove that all these changes are actually re-orders and there are no other differences.

With 20 files changed, I feel this is where we would ask for smaller PRs.

File: src/xdocs/checks/coding/multiplestringliterals.xml (hunk 1 above)
Comment (rnveach) [reply src/xdocs/checks/coding/multiplestringliterals.xml]: Leaving this open for others to see the bug.

**Comment by rnveach** [comment]
https://github.com/checkstyle/checkstyle/issues/13666 should be done first before adding the macro. The re-ordering should be a manual process of moving to a new position so we are sure new macro isn't causing issues.

**Comment by romani** [comment]
>there anyway you can prove that all these changes are actually re-orders and there are no other differences.

I think it easy to see in PR. I can confirm that I just moved block up/down in such files manually. Any misalignment or different order in files cause test failure. So it is self tested.

>

**Comment by romani** [comment]
>should be done first before adding the macro

Ordering is not a big deal. Ones all be on macros, changing order will be a way easier as all will be generated by macros. 
We need macros usage in all first we need to catch all other problems and edge cases.

**Comment by rnveach** [comment]
> I can confirm that I just moved block up/down in such files manually. 

If this was the case, then the link would not have changed so drastically or it would have been caught, unless it was seen and never questioned and just assumed to be correct. It is not easy for reviewers to see this 1 line change in a group of text change. GH doesn't even help us in this case with any highlighting.

> I see more value to use macros more widely to catch edge cases quicker.
> Ordering is not a big deal

It would be easier if macro was working correctly. It is now shown the macro is not that ready yet and still needs to be closely reviewed. It is easier for reviewers to split order and macro integration. This is why the other issue was created as there was agreement in PR I brought this up in.

**Comment by romani** [comment]
All such minor problems can be resolved later on if not noticed during review.

**Comment by romani** [comment]
I removed macroses, as keep only order changes

**Comment by rnveach** [comment]
https://github.com/checkstyle/checkstyle/pull/13717#issuecomment-1712534754
> is there anyway you can prove that all these changes are actually re-orders and there are no other differences

It makes me question this PR if we are just removing the macro and leaving in the changes that the macro made.

**Comment by romani** [comment]
There is no macros in this PR, just moving of content blocks in sync in all files

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java [hunk 2]
```diff
@@ -41,21 +41,21 @@
  * </p>
  * <ul>
  * <li>
- * Property {@code illegalClassNames} - Specify throw class names to reject.
- * Type is {@code java.lang.String[]}.
- * Default value is {@code Error, RuntimeException, Throwable, java.lang.Error,
- * java.lang.RuntimeException, java.lang.Throwable}.
+ * Property {@code ignoreOverriddenMethods} - Allow to ignore checking overridden methods
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java:44]: > just moving of content blocks in sync in all files.

If we are just moving and doing nothing else, then how did the `a` change from lowercase to capital?

Is there a way to prove that this is just re-ordering and no other changes?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java:44]: It is ordering and changing capital letters in message to simplify diff in macros.

There not that much places where lowercase is used. So we need to capitalize them to be consistent.

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java:44]: Issue just says ordering.

Also removal from `MODULES_WITH_UNORDERED_PROPERTIES` is missing.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java:44]: https://github.com/checkstyle/checkstyle/blob/f61e24307e6b77cb2aa705226f492257fca03654/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L230

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java:44]: I already forgot about this, let me see if this is same .

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java:44]: fixed.

**Comment by romani** [comment]
done at https://github.com/checkstyle/checkstyle/pull/13717

**Pull request closed without merging**